### ISSUE TITLE
Wrong naming of released text files

### ISSUE THREAD

**Issue by PranjaLBiswas27** [AlexeyAB/Yolo_mark#88]
Hi,

I am trying to label my datset, but I see that when the names released text files is not consistent with the name of my image. In the screenshot below I ran the Yolo_Mark using top left console. There it can be seen that the text file was named "10.txt" which was released for the 3rd image(index starting from 0). First 2 text files were name correctly. Also for the 4th image the released text file is "100.txt". Is there a bug or I am missing something. Please let me know. Thanks a lot in advance

Kind Regards
Pranjal Biswas 

![screenshot from 2018-09-07 11-33-10](https://user-images.githubusercontent.com/38991742/45211258-17427200-b292-11e8-88e8-bb1501344bb2.png)

**Comment by AlexeyAB** [comment]
For `100.jpg` file will be created `100.txt`.

----

> images are not scrolled in the alphanumeric order

Yes, it is ordered by OpenCV function [`cv::glob()`](https://docs.opencv.org/3.4.0/db/de0/group__core__utils.html#gaf91b7f383218e2d3f760939140942297): https://github.com/AlexeyAB/Yolo_mark/blob/102ec4bcd1c0b5e0b5f784a16db9ac7e676040cb/main.cpp#L177

But then it is reordered by using C++ `std::sort()`: https://github.com/AlexeyAB/Yolo_mark/blob/102ec4bcd1c0b5e0b5f784a16db9ac7e676040cb/main.cpp#L215-L217

May be I'll fix it.

**Comment by AlexeyAB** [comment]
> Thanks a lot, any expected time of resolving the issue?

While there are no forecasts.

> But all files are not scrolled and whatever 

What do you mean?
As I see you have 894 images. Do you actually have more than 894 images?

**Comment by PranjaLBiswas27** [comment]
Yes I have 894 images in the screenshot I posted. So the scrolling is in the following order:
0,1,10,100,101,102,103...
When I have just 9 images, the scroll is in the following order:
0,1,2,3,4,5,6,7,8
When It is 10 images, it is as follows:
0,1,10

Just an observation. I don know whats the issue though. So whenever there are more or equal to 10 images the scroll order becomes wrong.

**Comment by AlexeyAB** [comment]
You can scroll all the images, but the problem is only in the order of the images.

`0,1,10` It doesn't depends on number of images. This is simply because the ordering of strings in C ++ works in this way, for lines: `0` < `1` < `10` < `2`.
Because it is first compared the first character i.e. `0` < `1` <= `1` < `2`

**Comment by PranjaLBiswas27** [comment]
Ahh, I get the issue now. Thanks a lot though. Hope the issue gets resolved soon.
